src/main/java/eu/darkbot/api/objects/Gui.java

```diff
@@ -0,0 +1,33 @@
+package eu.darkbot.api.objects;
+
+public interface Gui extends Rectangle {
+    /**
+     * @return true if gui window is visible
+     */
+    boolean isVisible();
+
+    /**
+     * Returns true only if state of gui equals visible param
+     * and gui animation is done.
+     *
+     * Changing state of gui takes ~1 second.
+     *
+     * @param visible should gui be visible
+     * @return true if state of gui equals visible param
+     */
+    boolean setVisible(boolean visible);
+
+    /**
+     * Clicks at gui position + plusX/Y
+     */
+    void click(int plusX, int plusY);
+
+    /**
+     * Moves mouse at gui position + plusX/Y
+     */
+    void hover(int plusX, int plusY);
+
+    //should be available in API?
+    long getElementsList(int elementsListId);
+    //and so on
```

Comment:
definetly not, in fact the api should probably not assume "clicks" or anything in gui either, it should present higher-level apis, somewhat like pet that has setGear etc